examples/classification/benchmark.py

```diff
@@ -0,0 +1,377 @@
+    )
+
+    results = BenchmarkResults()
+    progress_bar = tqdm(total=config.num_iterations)
+
+    for iteration, batch in enumerate(data_loader):
+        iter_start = time.time()
+
+        # inference
+        _ = _run_model(args=config, model=model, batch=batch)
+
+        iter_end = time.time()
+
+        if iteration >= config.num_warmup_iterations:
+            results.append_batch(
+                time_start=iter_start,
+                time_end=iter_end,
+                batch_size=config.batch_size,
+            )
+            progress_bar.update(1)
+
+    progress_bar.close()
+    print(f"Benchmarking complete. End-to-end results:\n{results}")
+
+
+def fix_onnx_input_shape(
+    model_path: str,
+    image_shape: Optional[Tuple[int]],
+) -> Tuple[str, Optional[NamedTemporaryFile]]:
+    """
+    Creates a new ONNX model from the given path that accepts the given input
+    shape. If the given model already has the given input shape no modifications are
+    made. Uses a tempfile to store the modified model file.
+
+    :param model_path: file path to ONNX model or SparseZoo stub of the model
+        to be loaded
+    :param image_shape: 2-tuple of the image shape to resize this model to, or None if
+        no resizing needed
+    :return: filepath to an onnx model reshaped to the given input shape will be the
+        original path if the shape is the same.  Additionally returns the
+        NamedTemporaryFile for managing the scope of the object for file deletion.
+        Additionally returns the image-shape to benchmark the new model with.
+    """
+    original_model_path = model_path
+    if model_path.startswith("zoo:"):
+        # load SparseZoo Model from stub
+        model = Zoo.load_model_from_stub(model_path)
+        model_path = model.onnx_file.downloaded_path()
+        print(f"Downloaded {original_model_path} to {model_path}")
+
+    model = onnx.load(model_path)
+    model_input = model.graph.input[0]
+
+    original_x = get_tensor_dim_shape(model_input, 2)
+    original_y = get_tensor_dim_shape(model_input, 3)
+    original_image_shape = (original_x, original_y)
+
+    if image_shape is None or original_image_shape == tuple(image_shape):
+        return model_path, None, original_image_shape  # no shape modification needed
+
+    set_tensor_dim_shape(model_input, 2, image_shape[0])
+    set_tensor_dim_shape(model_input, 3, image_shape[1])
+
+    tmp_file = NamedTemporaryFile()  # file will be deleted after program exit
+    onnx.save(model, tmp_file.name)
+
+    print(
+        f"Overwriting original model shape {original_image_shape} to {image_shape}\n"
+        f"Original model path: {original_model_path}, new temporary model saved to "
+        f"{tmp_file.name}"
+    )
+
+    return tmp_file.name, tmp_file, image_shape
+
+
+def _run_model(
+    args, model: Any, batch: Union[numpy.ndarray]
+) -> List[Union[numpy.ndarray]]:
+    # run model according to engine type
+    if args.engine == ORT_ENGINE:
```

Comment:
technically we are incurring more of a hit for benchmarking ORT compared with deepsparse. It shouldn't make that much of a difference given the run time of the models, but in good faith for the user, it would be good if they saw us just timing the execution and not any array or dictionary manipulations. Can we change this method to return the batch and the time and each implementation can wrap just the engine execution time?

Reply:
👌  done